### PR TITLE
Allow multiple configs in single skaffold.yaml

### PR DESCRIPTION
**Pull request by gsquared94**
**Related**:  [Skaffold Multiple Configs support ](https://tinyurl.com/skaffold-multiple-configs)
This PR implements defining multiple skaffold pipelines in a single file. Maintaining multiple configurations is already implemented in #5160

### CODE REVIEW

File: pkg/skaffold/schema/versions.go [hunk 1]
```diff
@@ -237,3 +209,97 @@ func ParseConfigAndUpgrade(filename, toVersion string) ([]util.VersionedConfig,
 	}
 	return upgraded, nil
 }
+
+// configFactoryFromAPIVersion checks that all configs in the input stream have the same API version, and returns a function to create a config with that API version.
+func configFactoryFromAPIVersion(buf []byte) (func() util.VersionedConfig, error) {
+	// This is to quickly check that it's possibly a skaffold.yaml,
+	// without parsing the whole file.
+	if !bytes.Contains(buf, []byte("apiVersion")) {
```
Comment (nkubala) [inline pkg/skaffold/schema/versions.go:217]: i think before doing this, we might want to also check if the file contains `kind: Config` as a further safeguard against users accidentally providing k8s configs to skaffold

**Review by nkubala — DISMISSED** [review]
thinking about this a little more, i don't know that we want to require all configs passed in to *explicitly* have the same API version.

in most cases, we should be able to take the latest found API version, and `skaffold fix` up the older ones to that new version successfully. this won't always work (in the event that there was a breaking config change at some point), but these are usually rare so it should work in the majority of cases. WDYT?

**Comment by gsquared94** [comment]
> thinking about this a little more, i don't know that we want to require all configs passed in to _explicitly_ have the same API version.
> 
> in most cases, we should be able to take the latest found API version, and `skaffold fix` up the older ones to that new version successfully. this won't always work (in the event that there was a breaking config change at some point), but these are usually rare so it should work in the majority of cases. WDYT?

So this only limits all configs defined in the **same** `skaffold.yaml` file to have the same config version. Different files can have different config versions obviously. WDYT?

File: pkg/skaffold/schema/versions.go (hunk 1 above)
Comment (gsquared94) [reply pkg/skaffold/schema/versions.go:217]: while that makes sense, I don't want to add that as a new validation in this PR. This is just a refactor of existing code.

Comment (nkubala) [reply pkg/skaffold/schema/versions.go:217]: got it, thought you added this.

**Comment by nkubala** [comment]
> So this only limits all configs defined in the same skaffold.yaml file to have the same config version. Different files can have different config versions obviously. WDYT?

ahh ok, sorry i didn't follow that. i don't think it's a huge deal, but if different files can have different config versions, and then i can combine those separate files into a single (legal) skaffold.yaml which now will fail because the config versions are different, that feels a bit arbitrary no?

i won't die on this hill because it's pretty unconventional and might make the implementation more annoying for you since you'll have to figure out how to make multi-config skaffold.yamls work with `skaffold fix`, but this does feel a tad inconsistent.

**Comment by gsquared94** [comment]
> ahh ok, sorry i didn't follow that. i don't think it's a huge deal, but if different files can have different config versions, and then i can combine those separate files into a single (legal) skaffold.yaml which now will fail because the config versions are different, that feels a bit arbitrary no?
> i won't die on this hill because it's pretty unconventional and might make the implementation more annoying for you since you'll have to figure out how to make multi-config skaffold.yamls work with `skaffold fix`, but this does feel a tad inconsistent.

You're right. It now supports multiple config versions. It was mostly an arbitrary choice on my part and not implementation difficulty. Fixed!

**Review by MarlonGamez — APPROVED** [review]
This LGTM 👍🏼

**Pull request merged**